src/main/kotlin/org/rust/lang/core/lexer/RustStringLiteralLexer.kt

```diff
@@ -0,0 +1,129 @@
+package org.rust.lang.core.lexer
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.StringEscapesTokenTypes.*
+import com.intellij.psi.tree.IElementType
+
+const val BYTE_ESCAPE_LENGTH = "\\x00".length
+const val UNICODE_ESCAPE_MIN_LENGTH = "\\u{0}".length
+const val UNICODE_ESCAPE_MAX_LENGTH = "\\u{000000}".length
```

Comment:
Things are `public` by default in Kotlin. It's better to add `private` here (meaning, visible only in this file) and remove `public` from other places.